New Swift Runner is not rendering proper output

@kazk I have no idea what is going on but the latest runner is not sending proper output:

https://preview.codewars.com/kata/some-egyptian-fractions/train/swift

![image](https://user-images.githubusercontent.com/579835/28235581-aca5805c-68c5-11e7-8efd-40b547f38f1f.png)

I'm looking into it now to make sure there wasn't some sort of deployment error but I need to leave soon.

Ah, I completely missed that it was a breaking change to existing content. I had thought there was some sort of rewrite already happening. I see two options:

### Option 1
Automatically rewrite user code with `opts.fixture.replace(/^ *(XCTMain *\( *\[)/m, '_$1')`

### Option 2
Don't use an Observer at all and simply format the natural output, since it seems to be easily parsable:

```
Test Suite 'All tests' started at 02:03:29.842
Test Suite 'codewarrior.xctest' started at 02:03:29.998
Test Suite 'SolutionTest' started at 02:03:29.998
Test Case 'SolutionTest.decompse' started at 02:03:29.999
/workspace/main.swift:9: error: SolutionTest.decompse : XCTAssertEqual failed: ("test") is not equal to ("1/2,1/4") - should return 1/2,1/4
/workspace/main.swift:9: error: SolutionTest.decompse : XCTAssertEqual failed: ("test") is not equal to ("3") - should return 3
/workspace/main.swift:9: error: SolutionTest.decompse : XCTAssertEqual failed: ("test") is not equal to ("1/2,1/4,1/20") - should return 1/2,1/4,1/20
/workspace/main.swift:9: error: SolutionTest.decompse : XCTAssertEqual failed: ("test") is not equal to ("1/2,1/7,1/59,1/5163,1/53307975") - should return 1/2,1/7,1/59,1/5163,1/53307975
/workspace/main.swift:9: error: SolutionTest.decompse : XCTAssertEqual failed: ("test") is not equal to ("1/2,1/3,1/9,1/83,1/34362") - should return 1/2,1/3,1/9,1/83,1/34362
Test Case 'SolutionTest.decompse' failed (0.001 seconds)
Test Suite 'SolutionTest' failed at 02:03:30.000
	 Executed 1 test, with 5 failures (0 unexpected) in 0.001 (0.001) seconds
Test Suite 'codewarrior.xctest' failed at 02:03:30.000
	 Executed 1 test, with 5 failures (0 unexpected) in 0.001 (0.001) seconds
Test Suite 'All tests' failed at 02:03:30.000
	 Executed 1 test, with 5 failures (0 unexpected) in 0.001 (0.001) seconds
```

I'm going to go with option #1 right now since it seems safer and less error prone.

--- a/lib/runners/swift.js
+++ b/lib/runners/swift.js
@@ -52,7 +52,7 @@ module.exports.run = function run(opts, cb) {
         args.push('/workspace/setup.swift');
       }
       fs.writeFileSync('/workspace/solution.swift', opts.solution);
-      fs.writeFileSync('/workspace/main.swift', opts.fixture);
+      fs.writeFileSync('/workspace/main.swift', opts.fixture.replace(/^\s*XCTMain\(/m, '_XCTMain('));
       args.push('/workspace/solution.swift', '/workspace/main.swift');
 
       return exec(args.join(' '), (err, stdout, stderr) => {

--- a/lib/runners/swift.js
+++ b/lib/runners/swift.js
@@ -52,7 +52,7 @@ module.exports.run = function run(opts, cb) {
         args.push('/workspace/setup.swift');
       }
       fs.writeFileSync('/workspace/solution.swift', opts.solution);
-      fs.writeFileSync('/workspace/main.swift', opts.fixture.replace(/^\s*XCTMain\(/m, '_XCTMain('));
+      fs.writeFileSync('/workspace/main.swift', opts.fixture.replace(/^\s*XCTMain\s*\(/m, '_XCTMain('));
       args.push('/workspace/solution.swift', '/workspace/main.swift');
 
       return exec(args.join(' '), (err, stdout, stderr) => {

--- a/test/runners/swift_spec.js
+++ b/test/runners/swift_spec.js
@@ -87,6 +87,41 @@ describe('swift runner', function() {
       });
     });
 
+    it('should allow whitespace in "XCTMain ("', function(done) {
+      var code = `
+                    func add(_ a: Int, _ b: Int) -> Int {
+                      return a + b
+                    }
+                `;
+      runner.run({
+        language: 'swift',
+        languageVersion: '3',
+        code: code,
+        fixture: `
+                            import XCTest
+                            class AddTest: XCTestCase {
+                              static var allTests = [
+                                    ("testAddCheck", testAddCheck),
+                              ]
+                              func testAddCheck() {
+                                XCTAssertEqual(add(1, 1), 2, "add(1, 1) should be 2")
+                              }
+                            }
+
+                            XCTMain  ([
+                                testCase(AddTest.allTests)
+                            ])
+                        `,
+        testFramework: 'xctest'
+      }, function(buffer) {
+        expect(buffer.stdout).to.contain('<DESCRIBE::>AddTest');
+        expect(buffer.stdout).to.contain('<IT::>testAddCheck');
+        expect(buffer.stdout).to.contain('<PASSED::>Test Passed');
+        expect(buffer.stdout).to.contain('<COMPLETEDIN::>');
+        done();
+      });
+    });
+
     it('should replace newlines in error messages with <:LF:>', function(done) {
       var code = `
                     class Calculator {

--- a/test/runners/swift_spec.js
+++ b/test/runners/swift_spec.js
@@ -73,7 +73,7 @@ describe('swift runner', function() {
                               }
                             }
 
-                            _XCTMain([
+                            XCTMain([
                                 testCase(CalculatorTest.allTests)
                             ])
                         `,
@@ -113,7 +113,7 @@ describe('swift runner', function() {
                               }
                             }
 
-                            _XCTMain([
+                            XCTMain([
                                 testCase(CalculatorTest.allTests)
                             ])
                         `,
@@ -174,7 +174,7 @@ describe('swift runner', function() {
                               }
                             }
 
-                            _XCTMain([
+                            XCTMain([
                                 testCase(CalculatorTest.allTests)
                             ])
                         `,
@@ -227,7 +227,7 @@ describe('swift runner', function() {
                               }
                             }
 
-                            _XCTMain([
+                            XCTMain([
                                 testCase(CalculatorTest.allTests)
                             ])
                         `,
@@ -274,7 +274,7 @@ describe('swift runner', function() {
                               }
                             }
 
-                            _XCTMain([
+                            XCTMain([
                                 testCase(GreetingsTest.allTests)
                             ])
                         `,
@@ -332,7 +332,7 @@ describe('swift runner', function() {
                                 }
                             }
 
-                            _XCTMain([
+                            XCTMain([
                                 testCase(CalculatorTest.allTests),
                                 testCase(PersonTest.allTests),
                             ])

